File: array-sum/array-sum.js
var givenArray = [];
var sum = 0;

do {
    var userInput = prompt("Enter element (leave empty or enter incorrect number format to finish): ");
    var element = Number(userInput);
    if(element)
        givenArray.push(element);
} while(element)

for (var i =0; i<givenArray.length; i++) {
    sum += givenArray[i];
}

alert(sum);

File: array-sum/array-sum.cpp
#include <iostream>

using namespace std;

int main()
{
    double sum = 0;
    double input = 0;
    
    cout<<"Enter elements of array (input incorrect number or empty line to finish"<<endl;
   
    while(true)
    {
    	cout<<"element = ";
    	
    	if(cin>>input) sum+=input;
    	else break;
    }
    
    cout<<"Sum = "<<sum<<endl; 
    
    return 0;
}
